smithy-jsonschema/src/main/java/software/amazon/smithy/jsonschema/PropertyNamingStrategy.java

```diff
@@ -51,6 +52,11 @@ static PropertyNamingStrategy createMemberNameStrategy() {
      */
     static PropertyNamingStrategy createDefaultStrategy() {
         return (containingShape, member, config) -> {
+            // Use the property$name field with highest priority
+            if (member.getTrait(PropertyTrait.class).flatMap(PropertyTrait::getName).isPresent()) {
```

Comment:
Why would the resource property binding impact JSON Schema? Shouldn't it match the structure itself?

Reply:
Looks like this was an early modification holdover so I removed it. Everything still works because [there is code in CfnConverter](https://github.com/awslabs/smithy/blob/f44e2c1a86de7cca9c65475dc7a9cb6b8118c058/smithy-aws-cloudformation/src/main/java/software/amazon/smithy/aws/cloudformation/schema/fromsmithy/CfnConverter.java#L242) handling it before this point.